spectrum-offchain-lm/src/bundle/rocksdb.rs

```diff
@@ -0,0 +1,218 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use spectrum_offchain::{
+    binary::prefixed_key,
+    data::{
+        unique_entity::{Confirmed, Predicted, Traced, Unconfirmed},
+        OnChainEntity,
+    },
+};
+use tokio::task::spawn_blocking;
+
+use crate::data::{AsBox, BundleId, BundleStateId, PoolId};
+
+use super::{BundleRepo, EpochHeightRange, StakingBundle};
+
+pub struct BundleRepoRocksDB {
+    pub db: Arc<rocksdb::OptimisticTransactionDB>,
+    pub epoch_len: u32,
+}
+
+#[async_trait]
+impl BundleRepo for BundleRepoRocksDB {
+    fn get_epoch_length(&self) -> u32 {
+        self.epoch_len
+    }
+
+    fn get_height_range(&self, epoch_ix: u32) -> EpochHeightRange {
+        let first = (epoch_ix - 1) * self.epoch_len;
+        let last = epoch_ix * self.epoch_len - 1;
+        EpochHeightRange { first, last }
+    }
+
+    async fn select(&self, pool_id: PoolId, epoch_ix: u32) -> Vec<BundleId> {
+        assert!(epoch_ix > 0);
+
+        let EpochHeightRange { first, last } = self.get_height_range(epoch_ix);
+
+        let db = self.db.clone();
+
+        spawn_blocking(move || {
+            let mut bundle_ids = vec![];
+            for (_, value) in db
+                .iterator(rocksdb::IteratorMode::From(
+                    &bincode::serialize(STATE_PREFIX.as_bytes()).unwrap(),
+                    rocksdb::Direction::Forward,
+                ))
+                .flatten()
+            {
+                if let Ok(bundle) = bincode::deserialize::<'_, AsBox<StakingBundle>>(&value) {
+                    let bundle_height = bundle.0.creation_height;
```

Comment:
creation_height of the bundle box has nothing to do with epochs bundle is eligible for